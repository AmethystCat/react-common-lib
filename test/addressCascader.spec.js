import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AddressCascader from '../components/addressCascader';

Enzyme.configure({ adapter: new Adapter() });

describe('AddressCascader UT', () => {
  test('snapshot', () => {
    const component = renderer.create(<AddressCascader />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  
  test('should_call_onChange_when_select_address', () => {
    // given
    const onChange = jest.fn();
    const wrapper = shallow(<AddressCascader onChange={onChange} />);
    const instance = wrapper.instance();
    const code = ["11", "1101", "110101"]
    const expectArgs = {provinceCode: '11', districtCode: '110101', cityCode: '1101'};
    // when
    instance.onCascaderChange(code);
    // then 
    expect(onChange).toBeCalledWith(expectArgs);
  });
  
  test('should_value_fully_controlled_when_value_pass_dynamically', () => {
    // given
    const onChange = jest.fn();
    const value = {provinceCode: '12', cityCode: '1201', districtCode: '120101'};
    const code = ["11", "1101", "110101"];
    const expectState = {provinceCode: '11', cityCode: '1101', districtCode: '110101'};
    const wrapper = shallow(<AddressCascader onChange={onChange} />);
    const instance = wrapper.instance();
    // when
    wrapper.setProps({ value });
    const { state } = instance;
    // then
    expect(state).toEqual(value);
    
    // when
    instance.onCascaderChange(code);
    // then
    expect(instance.state).toEqual(value);
    expect(onChange).toBeCalledWith(expectState);
  });
});
