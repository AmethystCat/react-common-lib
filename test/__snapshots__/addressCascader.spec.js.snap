// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressCascader UT snapshot 1`] = `
<div
  className="section-address"
>
  <span
    className="ant-cascader-picker"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <span
      className="ant-cascader-picker-label"
    >
      
    </span>
    <input
      autoComplete="off"
      className="ant-input ant-cascader-input "
      disabled={false}
      onKeyDown={[Function]}
      placeholder="省 / 市 / 区"
      readOnly={true}
      type="text"
      value=""
    />
    <i
      className="anticon anticon-down ant-cascader-picker-arrow"
    />
  </span>
</div>
`;
